clang/lib/Lex/PPDirectives.cpp

```diff
@@ -3288,7 +3288,7 @@ void Preprocessor::HandleIfdefDirective(Token &Result,
     return;
   }
 
-  emitMacroExpansionWarnings(MacroNameTok);
+  emitMacroExpansionWarnings(MacroNameTok, true);
```

Comment:
```suggestion
  emitMacroExpansionWarnings(MacroNameTok, /*IsIfnDef=*/true);
```

Reply:
The bool argument is not needed since I am removing the 3rd argument of the diagnostic?

Reply:
You'll still need this one because that's what controls silencing the diagnostic in an ifndef.